layer/compile_time_layer.cc

```diff
@@ -40,10 +40,10 @@ constexpr char kLayerDescription[] =
 constexpr char kLogFilenameEnvVar[] = "VK_COMPILE_TIME_LOG";
 
 performancelayers::LayerData* GetLayerData() {
-  static performancelayers::LayerData* layer_data =
-      new performancelayers::LayerData(getenv(kLogFilenameEnvVar),
-                                       "Pipeline,Compile Time (ns)");
-  return layer_data;
+  // Don't use new -- make the destructor run when the layer gets unloaded.
+  static performancelayers::LayerData layer_data = performancelayers::LayerData(
+      getenv(kLogFilenameEnvVar), "Pipeline,Compile Time (ns)");
```

Comment:
why not used some C++ managed pointer and use RAII?

Reply:
What difference would it make here? I think the point of using new in the past was to not register a global destructor as suggested by the c++ style guide, but in this case we do want to run the destructor when the layer gets unloaded.

Reply:
IE the lifetime on the meta-level is managed by the Vulkan loader.

Reply:
ah, nm.  In that case, this is fine.  Thanks.